opsdroid/events.py

```diff
@@ -0,0 +1,166 @@
+        connector: Connector object used to interact with given chat service
+        text: String text of message
+        raw_message: Raw message as provided by chat service. None by default
+
+    Attributes:
+        created: Local date and time that message object was created
+        user: String name of user sending message
+        room: String name of the room or chat channel in which message was sent
+        connector: Connector object used to interact with given chat service
+        text: Text of message as string
+        raw_message: Raw message provided by chat service
+        regex: A re match object for the regular expression message was matched
+            against
+        responded_to: Boolean initialized as False. True if message has been
+            responded to
+
+    """
+
+    def __init__(self, user, room, connector,
+                 text, raw_message=None):  # noqa: D107
+        """Create object with minimum properties."""
+        super().__init__(user, room, connector)
+        self.text = text
+        self.raw_message = raw_message
+        self.regex = None
+
+    async def _thinking_delay(self):
+        """Make opsdroid wait x-seconds before responding.
+
+        Number of seconds defined in YAML config. file, accessed via connector.
+        """
+        seconds = self.connector.configuration.get('thinking-delay', 0)
+
+        if isinstance(seconds, list):
+            seconds = randrange(seconds[0], seconds[1])
+
+        await asyncio.sleep(seconds)
+
+    async def _typing_delay(self, text):
+        """Delays reply to simulate typing.
+
+        Seconds to delay equals number of characters in response multiplied by
+        number of seconds defined in YAML config. file, accessed via connector.
+        """
+        seconds = self.connector.configuration.get('typing-delay', 0)
+        char_count = len(text)
+
+        if isinstance(seconds, list):
+            seconds = randrange(seconds[0], seconds[1])
+
+        await asyncio.sleep(char_count*seconds)
+
+    async def respond(self, response_event, room=None):
+        """Respond to this message using the connector it was created by.
+
+        Creates copy of this message with updated text as response.
+        Delays message if thinking or typing delay present in config. file.
+        Updates responded_to attribute to True if False.
+        Logs response and response time in OpsDroid object stats.
+        """
+        opsdroid = get_opsdroid()
+        if isinstance(response_event, str):
+            response = copy(self)
+            response.text = response_event
+        else:
+            response = response_event
+
+        if 'thinking-delay' in self.connector.configuration or \
+           'typing-delay' in self.connector.configuration:
+            await self._thinking_delay()
+            await self._typing_delay(response.text)
+
+        await self.connector.respond(response, room)
```

Comment:
If we are refactoring as I propose in #719 this should be `opdroid.send`.

Reply:
I agree.